### DISCUSSION TITLE
About rf.signal unit

### DISCUSSION THREAD

**Discussion by timnotavailable** [pulseq/pulseq#73] [Q&A]
Hello,
 about rf pulse event there is an attribute called signal, I would like to ask what is the unit of this attribute, and how the flip angle is calculated in Pulseq since the \gamma is not specified.
 I would like to define a pulse in arbitrary pulse module, but I don't know what scale should I give in the attribute signal , and it demands me to provide a flip angle.

**Comment by QingpingChen1201** [comment]
Hi,
many thanks for your message and your interest in using Pulseq! 
If you don't specify the gamma by using `system=mr.opt(...,'gamma', yourValue, ...)`, Pulseq will use the default gamma value of hydrogen: 42576000 Hz/T, which you can check by running `system.gamma` in Matlab.
However, the unit of rf.signal is not determined by the above gamma. If you have a look at the simpliest function for generating hard RF pulses (`mr.makeBlockPulse`), you will see this line `signal = opt.flipAngle/(2*pi)/opt.duration*ones(size(t));`. The corresponding equation is `signal=flipAngle/(2Xpi)/RFduration`. In Pulseq, the flip angle is given in **radian** (e.g., for a 90-degree RF pulse, flipAngle = pi/2 radians). Therefore, flipAngle/(2Xpi) has a unit of **circle**. This indicates that the unit of rf.signal is **circle/t**.
For example, you would like to design an arbitrary pulse whose flip angle is 90 degree or pi/2 radians. Then, the integral of the `signal` of this RF pulse over time should be 1/4 circle.
Hope this helps!
Please feel free to let us know if you have any further questions or need any further information.
Best regards,
Qingping
P.S. this is not an issue but a question, so I will move it to the Q&A session:-)

**Reply by Jody-Todd** [reply]
Is there a convenient way to to convert the rf signal to physical units (e.g. T)? Can I use mr.convert(rf.signal, 'Hz', 
'T')?

**Reply by m-a-x-i-m-z** [reply]
yes, we use it in the other direction when setting the RF amplitude limits in the system properties, so the conversion should also work for you to go from Hz to for instance uT.
--M
